api/src/main/java/org/openmrs/module/appointments/notification/impl/DefaultSmsSender.java

```diff
@@ -0,0 +1,180 @@
+     * @param as
+     * @return
+     */
+    private Properties smsSessionPropertiesFromOMRS() {
+        Properties p = new Properties();
+        p.put("sms.key.username", administrationService.getGlobalProperty("sms.key.username", ""));
+        p.put("sms.key.password", administrationService.getGlobalProperty("sms.key.password", ""));
+        p.put("sms.key.phonenumber", administrationService.getGlobalProperty("sms.key.phonenumber", ""));
+        p.put("sms.key.message", administrationService.getGlobalProperty("sms.key.message", ""));
+        p.put("sms.key.sender", administrationService.getGlobalProperty("sms.key.sender", ""));
+        p.put("sms.key.entity.identifier", administrationService.getGlobalProperty("sms.key.entity.identifier", ""));
+        p.put("sms.key.template.identifier", administrationService.getGlobalProperty("sms.key.template.identifier", ""));
+
+        p.put("sms.url", administrationService.getGlobalProperty("sms.url", ""));
+        p.put("sms.username", administrationService.getGlobalProperty("sms.username", ""));
+        p.put("sms.password", administrationService.getGlobalProperty("sms.password", ""));
+        p.put("sms.sender", administrationService.getGlobalProperty("sms.sender", ""));
+        p.put("sms.entity.identifier", administrationService.getGlobalProperty("sms.entity.identifier", ""));
+        p.put("sms.appointment.walkin.template.identifier", administrationService.getGlobalProperty("sms.appointment.walkin.template.identifier", ""));
+        p.put("sms.appointment.teleconsultation.template.identifier", administrationService.getGlobalProperty("sms.appointment.teleconsultation.template.identifier", ""));
+        p.put("sms.appointment.walkin.message.template", administrationService.getGlobalProperty("sms.appointment.walkin.message.template", ""));
+        p.put("sms.appointment.teleconsultation.message.template", administrationService.getGlobalProperty("sms.appointment.teleconsultation.message.template", ""));
+        p.put("sms.params.addons", administrationService.getGlobalProperty("sms.params.addons", ""));
+        return p;
+    }
+
+    private Properties smsSessionPropertiesFromPath() {
+        Path propertyFilePath = Paths.get(OpenmrsUtil.getApplicationDataDirectory(), SMS_PROPERTIES_FILENAME);
+        if (Files.exists(propertyFilePath)) {
+            Properties properties = new Properties();
+            try {
+                log.info("Reading properties from: " + propertyFilePath);
+                properties.load(Files.newInputStream(propertyFilePath));
+                return properties;
+            } catch (IOException e) {
+                log.error("Could not load sms properties from: " + propertyFilePath, e);
+            }
+        } else {
+            log.warn("No sms configuration defined at " + propertyFilePath);
+        }
+        return null;
+    }
+
+    private String getDateWithSuffix(int date) {
```

Comment:
do we really need to hardcode st, nd etc .. why not just convert to string representation of date .. like 30 June 2022. its also not using the locale and using just english. You can leverage a date formatter

Reply:
sure. Will incorporate all the comments.